core/src/main/java/org/infinispan/commands/functional/ReadWriteKeyCommand.java

```diff
@@ -130,4 +128,26 @@ public void init(ComponentRegistry componentRegistry) {
          ((InjectableComponent) f).inject(componentRegistry);
       }
    }
+
+   @Override
+   public final boolean isReturnValueExpected() {
+      return true;
+   }
+
+   @Override
+   public Metadata getMetadata() {
```

Comment:
TBH, I find this Metaparams stuff a bit complicated for what it really means and it is. In merge I just needed to pass the object like that and this was fine. If we are about to consider replacing other commands for the functional commands, honestly, I would consider using the metaparams stuff for the user but also the metadata object like that